hazelcast-client/src/main/java/com/hazelcast/client/cache/impl/AbstractClientInternalCacheProxy.java

```diff
@@ -157,13 +159,17 @@ protected void onInitialize() {
     public void setCacheManager(HazelcastCacheManager cacheManager) {
         assert cacheManager instanceof HazelcastClientCacheManager;
 
-        this.cacheManager = (HazelcastClientCacheManager) cacheManager;
+        if (!cacheManagerRef.compareAndSet(null, (HazelcastClientCacheManager) cacheManager)) {
+            if (cacheManagerRef.get() != cacheManager) {
```

Comment:
Not really, this code path is invoked both when creating and when getting a `Cache` so under normal usage it is most probable that the `cacheManager` is already set to the proper one --> no exception should be thrown.